model/task/task.go

```diff
@@ -2920,31 +2920,31 @@ func GetTasksByVersion(versionID string, opts GetTasksByVersionOptions) ([]Task,
 			},
 		})
 	}
-	countPipeline := []bson.M{}
-	countPipeline = append(countPipeline, pipeline...)
-	countPipeline = append(countPipeline, bson.M{"$count": "count"})
+
+	sortAndPaginatePipeline := []bson.M{}
```

Comment:
I think this function in general needs many more comments so that people besides you (i.e. me, gulp) can understand this in the future. To that end, can you drop a comment here explaining why you're starting a new pipeline?